test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/RegressionTest.scala

```diff
@@ -440,7 +442,15 @@ class RegressionTest {
   @Test def null_asInstanceOf_Unit_should_succeed_issue_1691(): Unit = {
     def getNull(): Any = null
     val x = getNull().asInstanceOf[Unit]: Any
```

Comment:
While you are here, can you comment why we have this contraption? (I guess to test both a potential constant folder in the optimizer and the runtime libs).